src/lang/modifyAst/addFillet.ts

```diff
@@ -89,30 +95,45 @@ export function modifyAstCloneWithFilletAndTag(
     )
     if (err(tagResult)) return tagResult
     const { tag } = tagResult
+    const tagInfo = { tag, selectionType }
 
     // Group tags by their corresponding extrude node
     const extrudeKey = JSON.stringify(pathToExtrudeNode)
 
     if (lookupMap.has(extrudeKey)) {
       const existingPath = lookupMap.get(extrudeKey)
       if (!existingPath) return new Error('Path to extrude node not found.')
-      extrudeToTagsMap.get(existingPath)?.push(tag)
+      extrudeToTagsMap.get(existingPath)?.push(tagInfo)
     } else {
       lookupMap.set(extrudeKey, pathToExtrudeNode)
-      extrudeToTagsMap.set(pathToExtrudeNode, [tag])
+      extrudeToTagsMap.set(pathToExtrudeNode, [tagInfo])
     }
   }
 
   // Step 2: Apply fillet(s) for each extrude node (body)
   let pathToFilletNodes: Array<PathToNode> = []
-  for (const [pathToExtrudeNode, tags] of extrudeToTagsMap.entries()) {
+  for (const [pathToExtrudeNode, tagInfos] of extrudeToTagsMap.entries()) {
     // Create a fillet expression with multiple tags
     const radiusValue =
       'variableName' in radius ? radius.variableIdentifierAst : radius.valueAst
+
+    const tagCalls = tagInfos.map(({ tag, selectionType }) => {
+      let tagCall: Expr = createIdentifier(tag)
+
+      // Modify the tag based on selectionType
+      if (selectionType === 'edge') {
+        tagCall = createCallExpressionStdLib('getOppositeEdge', [tagCall])
+      } else if (selectionType === 'adjacent-edge') {
+        tagCall = createCallExpressionStdLib('getNextAdjacentEdge', [tagCall])
+      }
+      return tagCall
```

Comment:
This pattern is duplicated in `hasValidFilletSelection()`. It's probably worth pulling it out to a function.

Reply:
Good point , thanks  bead90d4a2984e1600ba436ab08ad9be399af9a6